streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredTimestampedKeyValueStore.java

```diff
@@ -50,6 +50,31 @@
         super(inner, metricScope, time, keySerde, valueSerde);
     }
 
+    private MeteredTimestampedKeyValueStore(final KeyValueStore<Bytes, byte[]> inner,
+                                            final String metricsScope,
+                                            final String threadId,
+                                            final Time time,
+                                            final Serde<K> keySerde,
+                                            final Serde<ValueAndTimestamp<V>> valueSerde) {
+        super(inner, metricsScope, threadId, time, keySerde, valueSerde);
+    }
+
+    @Override
+    public MeteredTimestampedKeyValueStore<K, V> reWrap(final KeyValueStore<Bytes, byte[]> inner) {
```

Comment:
Overriding this here lets us preserve the correct type of the store when re-wrapping.